File: Chapter 04/JS/section_4_4_anti_patterns/after_rich_domain_thin_controller/business_logic/orderRequest.js
// Chapter 04/JavaScript/4.4 Anti-Patterns/After-RichDomain/business_logic/orderRequest.js

/**
 * DTO (Data Transfer Object) for incoming requests.
 */
class OrderRequest {
    /**
     * @param {number} customerId
     * @param {Array<import('../domain_models/item')>} items
     */
    constructor(customerId, items) {
        this.customerId = customerId;
        this.items = items;
    }
}

module.exports = OrderRequest;

File: Chapter 04/Python/4.4_anti_patterns/after_rich_domain_thin_controller/business_logic/order_request.py
# Chapter 04/Python/4.4 Anti-Patterns/After-RichDomain/business_logic/order_request.py
from typing import List
from ..domain_models.item import Item

class OrderRequest:
    """
    DTO (Data Transfer Object) for incoming requests.
    """
    def __init__(self, customer_id: int, items: List[Item]):
        self.customer_id = customer_id
        self.items = items